python/cudf/cudf/tests/test_options.py

```diff
@@ -45,33 +46,74 @@ def validator(x):
-
-
-def test_option_set_invalid(odd_option):
-    with pytest.raises(ValueError, match="Invalid option value 0"):
-        cudf.set_option("odd_option", 0)
-
-
-def test_option_description(odd_option):
-    s = StringIO()
-    with redirect_stdout(s):
-        cudf.describe_option("odd_option")
-    s.seek(0)
-    expected = "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
-    assert expected == s.read()
-
-
-def test_option_description_all(odd_option, even_option):
-    s = StringIO()
-    with redirect_stdout(s):
-        cudf.describe_option()
-    s.seek(0)
-    expected = (
-        "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
-        "even_option:\n\tAn even option.\n\t[Default: 0] [Current: 0]\n"
-    )
-    assert expected == s.read()
+@pytest.mark.usefixtures("odd_option", "even_option")
+class TestCleanOptions:
+    def test_option_get_set(odd_option):
+        assert cudf.get_option("odd_option") == 1
+        cudf.set_option("odd_option", 101)
+        assert cudf.get_option("odd_option") == 101
+
+    def test_option_set_invalid(odd_option):
+        with pytest.raises(ValueError, match="Invalid option value 0"):
+            cudf.set_option("odd_option", 0)
+
+    def test_option_description(odd_option):
+        s = StringIO()
+        with redirect_stdout(s):
+            cudf.describe_option("odd_option")
+        s.seek(0)
+        expected = (
+            "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
+        )
+        assert expected == s.read()
+
+    def test_option_description_all(odd_option, even_option):
+        s = StringIO()
+        with redirect_stdout(s):
+            cudf.describe_option()
+        s.seek(0)
+        expected = (
+            "odd_option:\n\tAn odd option.\n\t[Default: 1] [Current: 1]\n"
+            "even_option:\n\tAn even option.\n\t[Default: 0] [Current: 0]\n"
+        )
+        assert expected == s.read()
+
+
+def test_empty_option_context():
+    prev_setting = cudf.get_option("default_integer_bitwidth")
+    cudf.set_option("default_integer_bitwidth", 32)
+    with cudf.option_context():
+        assert cudf.get_option("default_integer_bitwidth") == 32
+        assert cudf.get_option("default_float_bitwidth") is None
+        assert cudf.get_option("spill") is os.getenv("CUDF_SPILL", False)
+        assert cudf.get_option("copy_on_write") is os.getenv(
+            "CUDF_COPY_ON_WRITE", False
+        )
+        assert cudf.get_option("mode.pandas_compatible") is False
+    assert cudf.get_option("default_integer_bitwidth") == 32
+    cudf.set_option("default_integer_bitwidth", prev_setting)
+
+
+def test_option_context():
+    prev_setting = cudf.get_option("mode.pandas_compatible")
+    cudf.set_option("mode.pandas_compatible", True)
+    assert cudf.get_option("mode.pandas_compatible") is True
+    with cudf.option_context(
+        "mode.pandas_compatible", False, "default_integer_bitwidth", 32
```

Comment:
Same comment about default integer bitwidth here too. Can we use a value that is guaranteed not the same as the one that is set (e.g. by saying `old_val = get_option(...)` `option_context(..., old_val + 1)`

Reply:
Done 👍